File: lib-commons/src/main/java/com/dataflow/mappers/AbstractBasicProtoMapper.java
package com.dataflow.mappers;

import com.google.protobuf.Timestamp;
import java.time.LocalDateTime;
import java.time.ZoneOffset;
import java.util.UUID;

@Deprecated // Use kotlin version in this lib
public interface AbstractBasicProtoMapper {

  //
  // String <-> UUID
  //

  default String map(UUID value) {
    return value.toString();
  }

  default UUID map(String value) {
    return UUID.fromString(value);
  }

  //
  // com.google.protobuf.Timestamp <-> LocalDateTime
  //

  default Timestamp map(LocalDateTime value) {
    return Timestamp.newBuilder()
        .setSeconds(value.toEpochSecond(ZoneOffset.UTC))
        .build();
  }

  default LocalDateTime map(Timestamp value) {
    return LocalDateTime.ofEpochSecond(value.getSeconds(), 0, ZoneOffset.UTC);
  }

}

File: lib-commons/src/main/kotlin/dataflow/commons/mappers/AbstractBasicProtoMapper.kt
package dataflow.commons.mappers

import com.google.protobuf.Timestamp
import java.time.LocalDateTime
import java.time.ZoneOffset
import java.util.*

abstract class AbstractBasicProtoMapper {

    //
    // String <-> UUID
    //

    fun map(value: UUID): String? {
        return value.toString()
    }

    fun map(value: String?): UUID? {
        return UUID.fromString(value)
    }

    //
    // com.google.protobuf.Timestamp <-> LocalDateTime
    //

    fun map(value: LocalDateTime): Timestamp? {
        return Timestamp.newBuilder()
            .setSeconds(value.toEpochSecond(ZoneOffset.UTC))
            .build()
    }

    fun map(value: Timestamp): LocalDateTime? {
        return LocalDateTime.ofEpochSecond(value.seconds, 0, ZoneOffset.UTC)
    }

}